### PR TITLE
Make Chef Infra Server honor license acceptance

### PR DESCRIPTION
**Pull request by bkonick**
Pulled this in from chef-cookbooks/chef-ingredient#241 since the original author hasn't been able to sign off on the commits and I want to upgrade to Chef Infra Server 13.

There are unrelated lint issues when I run `delivery local all`.

### Description

Pass license acceptance to `chef-server-ctl` when necessary.

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: resources/chef_ingredient.rb [hunk 1]
```diff
@@ -103,8 +103,15 @@
       end
     end
 
+    reconfigure_command = case new_resource.accept_license && new_resource.product_name == 'chef-server'
```
Comment (tas50) [inline resources/chef_ingredient.rb:106]: This seems better as a method in the action_class

```ruby
def reconfigure_command
  if new_resource.accept_license && new_resource.product_name == 'chef-server'
    'reconfigure --chef-license=accept'
  else
    'reconfigure'
  end
end
```

Comment (bkonick) [reply resources/chef_ingredient.rb]: Thanks for the suggestion. Just updated my branch.

**Comment by riccalioliojr** [comment]
Any updates on this?

**Comment by tas50** [comment]
Thanks @bkonick

**Pull request merged**